lib/class-wp-rest-menu-items-controller.php

```diff
@@ -516,135 +657,160 @@ public function get_item_schema() {
-			'type'        => 'integer',
-			'context'     => array( 'edit' ),
-			'default'     => 0,
-		);
-
 		$schema['properties']['type_label'] = array(
 			'description' => __( 'Name of type.' ),
 			'type'        => 'string',
-			'context'     => array( 'view', 'embed' ),
+			'context'     => array( 'view', 'edit', 'embed' ),
 			'readonly'    => true,
 		);
 
 		$schema['properties']['type'] = array(
-			'description' => __( 'Type of menu item' ),
+			'description' => __( 'The family of objects originally represented, such as "post_type" or "taxonomy".' ),
 			'type'        => 'string',
 			'enum'        => array( 'taxonomy', 'post_type', 'post_type_archive', 'custom' ),
 			'context'     => array( 'view', 'edit', 'embed' ),
+			'default'     => 'custom',
 		);
 
 		$schema['properties']['status'] = array(
 			'description' => __( 'A named status for the object.' ),
 			'type'        => 'string',
 			'enum'        => array_keys( get_post_stati( array( 'internal' => false ) ) ),
-			'context'     => array( 'view', 'edit' ),
-		);
-
-		$schema['properties']['link'] = array(
-			'description' => __( 'URL to the object.' ),
-			'type'        => 'string',
-			'format'      => 'uri',
+			'default'     => 'publish',
 			'context'     => array( 'view', 'edit', 'embed' ),
-			'readonly'    => true,
 		);
 
 		$schema['properties']['parent'] = array(
 			'description' => __( 'The ID for the parent of the object.' ),
 			'type'        => 'integer',
-			'context'     => array( 'view', 'edit' ),
+			'minimum'     => 0,
+			'default'     => 0,
+			'context'     => array( 'view', 'edit', 'embed' ),
 		);
 
 		$schema['properties']['attr_title'] = array(
-			'description' => __( 'The title attribute of the link element for this menu item .' ),
-			'context'     => array( 'view', 'edit' ),
+			'description' => __( 'Text for the title attribute of the link element for this menu item.' ),
 			'type'        => 'string',
+			'context'     => array( 'view', 'edit', 'embed' ),
+			'arg_options' => array(
+				'sanitize_callback' => 'sanitize_text_field',
+			),
 		);
-		$schema['properties']['classes']    = array(
-			'description' => __( 'The array of class attribute values for the link element of this menu item .' ),
-			'context'     => array( 'view', 'edit' ),
+
+		$schema['properties']['classes'] = array(
+			'description' => __( 'Class names for the link element of this menu item.' ),
 			'type'        => 'array',
 			'items'       => array(
 				'type' => 'string',
 			),
-		);
-
-		$schema['properties']['db_id'] = array(
-			'description' => __( 'The DB ID of this item as a nav_menu_item object, if it exists( 0 if it doesn\'t exist).' ),
-			'context'     => array( 'view', 'edit' ),
-			'type'        => 'integer',
+			'context'     => array( 'view', 'edit', 'embed' ),
+			'arg_options' => array(
+				'sanitize_callback' => function ( $value ) {
+					return array_map( 'sanitize_html_class', explode( ' ', $value ) );
```

Comment:
Since the classes is an array of strings, I don't think this `sanitize_callback` makes sense anymore. I think it needs to be `wp_parse_list` instead of the `explode`?

Reply:
`sanitize_html_class` function checks that the string is in the correct format. This makes sure that no special / breaking characters are in the string. I think this should remain.

Reply:
This line was basically copied from existing validation found in [core](https://github.com/WordPress/WordPress/blob/f545bb3f634c8e949db9051bd88924df592c3337/wp-includes/nav-menu.php#L559).

Reply:
Yep I agree the validation should stay. I’m just talking about building the list of strings to validate.

Reply:
Changed to use `wp_parse_list` -  https://github.com/WP-API/menus-endpoints/pull/31/commits/aa5242a53ada635a5be893fe7928bd0d2dc26cc0